platform/ui/src/components/SegmentationGroupTable/SegmentationGroupSegment.tsx

```diff
@@ -21,200 +21,159 @@ const SegmentItem = ({
-              {!disableEditing && (
-                <Icon
-                  name="row-edit"
-                  className={classnames('h-5 w-5', {
-                    'text-white': isLocked,
-                    'text-primary-light': !isLocked,
-                  })}
-                  onClick={e => {
-                    e.stopPropagation();
-                    onEdit(segmentationId, segmentIndex);
-                  }}
-                />
-              )}
-              {isVisible ? (
-                <Icon
-                  name="row-hide"
-                  className={classnames('h-5 w-5', {
-                    'text-white': isLocked,
-                    'text-primary-light': !isLocked,
-                  })}
-                  onClick={e => {
-                    e.stopPropagation();
-                    onToggleVisibility(segmentationId, segmentIndex);
-                  }}
-                />
-              ) : (
-                <Icon
-                  name="row-unhide"
-                  className={classnames('h-5 w-5', {
-                    'text-white': isLocked,
-                    'text-primary-light': !isLocked,
-                  })}
-                  onClick={e => {
-                    e.stopPropagation();
-                    onToggleVisibility(segmentationId, segmentIndex);
-                  }}
-                />
-              )}
-            </div>
-          )}
-        </div>
-      </div>
-      {/* <div className="relative flex flex-col w-full border-t border-t-black">
-        <div className="fl`ex items-center mb-1 ml-2">
-          <div className="flex items-center flex-1 text-base text-primary-light">
+        <div className="group-hover/row:bg-primary-dark flex h-full w-full flex-grow items-center">
+          <div className="pl-2 pr-1.5">
             <div
-              className={classnames(
-                'w-3 h-3 mt-1 mr-2 rounded-full cursor-pointer transition duration-300 hover:opacity-80'
-              )}
+              className={classnames('h-[8px] w-[8px] grow-0 rounded-full', {
+                'hover:cursor-pointer hover:opacity-60': !disableEditing,
+              })}
+              style={{ backgroundColor: cssColor }}
               onClick={e => {
+                if (disableEditing) {
+                  return;
+                }
                 e.stopPropagation();
                 onColor(segmentationId, segmentIndex);
               }}
-              style={{ backgroundColor: cssColor }}
-            ></div>
-            {label}
+            />
           </div>
-          <div className="flex items-center w-1/3">
-            <div className="px-1">
+          <div className="flex items-center py-1 hover:cursor-pointer">{label}</div>
+        </div>
+        <div
+          className={classnames(
+            'absolute right-0 top-0 flex flex-row-reverse rounded-lg pr-[8px] pt-[3px]',
+            {}
+          )}
+        >
+          {!isVisible && !isRowHovering && (
```

Comment:
It is a bit confusing that each icon is rendered here when NOT hovering and then rendered on line 131 when hovering. It would be great if we only had to "look" in one place for each icon. But if it is too much trouble, then that is ok.

Reply:
seems like there is another bug

![CleanShot 2023-09-14 at 13 10 34](https://github.com/OHIF/Viewers/assets/7490180/ee6212c7-09ef-44c3-be20-ae578f3aa1e0)